Index: trunk-2.6.39/include/image.mk
===================================================================
--- trunk-2.6.39/include/image.mk	(revision 27017)
+++ trunk-2.6.39/include/image.mk	(working copy)
@@ -56,7 +56,7 @@
 	echo -ne '\xde\xad\xc0\xde' >> $(1)
 endef
 
-# pad to 4k, 8k, 64k, 128k and add jffs2 end-of-filesystem mark
+# pad to 4k, 8k, 64k, 128k, 256k and add jffs2 end-of-filesystem mark
 define prepare_generic_squashfs
 	dd if=$(1) of=$(KDIR)/tmpfile.0 bs=4k conv=sync
 	$(call add_jffs2_mark,$(KDIR)/tmpfile.0)
@@ -64,7 +64,9 @@
 	$(call add_jffs2_mark,$(KDIR)/tmpfile.1)
 	dd if=$(KDIR)/tmpfile.1 of=$(KDIR)/tmpfile.2 bs=64k conv=sync
 	$(call add_jffs2_mark,$(KDIR)/tmpfile.2)
-	dd if=$(KDIR)/tmpfile.2 of=$(1) bs=64k conv=sync
+	dd if=$(KDIR)/tmpfile.2 of=$(KDIR)/tmpfile.3 bs=64k conv=sync
+	$(call add_jffs2_mark,$(KDIR)/tmpfile.3)
+	dd if=$(KDIR)/tmpfile.3 of=$(1) bs=128k conv=sync
 	$(call add_jffs2_mark,$(1))
 	rm -f $(KDIR)/tmpfile.*
 endef
