Index: backfire/include/image.mk
===================================================================
--- backfire/include/image.mk	(revision 25795)
+++ backfire/include/image.mk	(working copy)
@@ -36,7 +36,7 @@
   SQUASHFS_OPTS  := -comp lzma -processors 1
 endif
 
-JFFS2_BLOCKSIZE ?= 64k 128k
+JFFS2_BLOCKSIZE ?= 64k 128k 256k
 
 define add_jffs2_mark
 	echo -ne '\xde\xad\xc0\xde' >> $(1)
@@ -50,7 +50,9 @@
 	$(call add_jffs2_mark,$(KDIR)/tmpfile.1)
 	dd if=$(KDIR)/tmpfile.1 of=$(KDIR)/tmpfile.2 bs=64k conv=sync
 	$(call add_jffs2_mark,$(KDIR)/tmpfile.2)
-	dd if=$(KDIR)/tmpfile.2 of=$(1) bs=64k conv=sync
+	dd if=$(KDIR)/tmpfile.2 of=$(KDIR)/tmpfile.3 bs=64k conv=sync
+	$(call add_jffs2_mark,$(KDIR)/tmpfile.3)
+	dd if=$(KDIR)/tmpfile.3 of=$(1) bs=128k conv=sync
 	$(call add_jffs2_mark,$(1))
 	rm -f $(KDIR)/tmpfile.*
 endef
Index: backfire/include/download.mk
===================================================================
--- backfire/include/download.mk	(revision 25795)
+++ backfire/include/download.mk	(working copy)
@@ -88,7 +88,7 @@
 		rm -rf $(SUBDIR) && \
 		[ \! -d $(SUBDIR) ] && \
 		git clone $(URL) $(SUBDIR) && \
-		(cd $(SUBDIR) && git checkout $(VERSION)) && \
+		(cd $(SUBDIR) && git checkout -b $(SUBDIR) $(VERSION)) && \
 		echo "Packing checkout..." && \
 		rm -rf $(SUBDIR)/.git && \
 		$(call dl_pack,$(TMP_DIR)/dl/$(FILE),$(SUBDIR)) && \
